src/semver/_deprecated.py

```diff
@@ -0,0 +1,378 @@
+        of your choice. By default, it's  :class:`DeprecationWarning`, but
+        you can choose :class:`PendingDeprecationWarning` or a custom class.
+    :return: decorated function which is marked as deprecated
+    """
+
+    if func is None:
+        return partial(deprecated, replace=replace, version=version, category=category)
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> Callable[..., F]:
+        msg_list = ["Function 'semver.{f}' is deprecated."]
+
+        if version:
+            msg_list.append("Deprecated since version {v}. ")
+        msg_list.append("This function will be removed in semver 3.")
+        if replace:
+            msg_list.append("Use {r!r} instead.")
+        else:
+            msg_list.append("Use the respective 'semver.VersionInfo.{r}' instead.")
+
+        f = cast(F, func).__qualname__
+        r = replace or f
+
+        frame = cast(FrameType, cast(FrameType, inspect.currentframe()).f_back)
+
+        msg = " ".join(msg_list)
+        warnings.warn_explicit(
+            msg.format(f=f, r=r, v=version),
+            category=category,
+            filename=inspect.getfile(frame.f_code),
+            lineno=frame.f_lineno,
+        )
+        # As recommended in the Python documentation
+        # https://docs.python.org/3/library/inspect.html#the-interpreter-stack
+        # better remove the interpreter stack:
+        del frame
+        return func(*args, **kwargs)  # type: ignore
+
+    return wrapper
+
+
+@deprecated(version="2.10.0")
+def parse(version):
+    """
+    Parse version to major, minor, patch, pre-release, build parts.
+
+    .. deprecated:: 2.10.0
+       Use :func:`semver.VersionInfo.parse` instead.
+
+    :param version: version string
+    :return: dictionary with the keys 'build', 'major', 'minor', 'patch',
+             and 'prerelease'. The prerelease or build keys can be None
+             if not provided
+    :rtype: dict
+
+    >>> ver = semver.parse('3.4.5-pre.2+build.4')
+    >>> ver['major']
+    3
+    >>> ver['minor']
+    4
+    >>> ver['patch']
+    5
+    >>> ver['prerelease']
+    'pre.2'
+    >>> ver['build']
+    'build.4'
+    """
+    return VersionInfo.parse(version).to_dict()
+
+
+@deprecated(replace="semver.VersionInfo.parse", version="2.10.0")
+def parse_version_info(version):
+    """
+    Parse version string to a VersionInfo instance.
+
+    .. deprecated:: 2.10.0
+       Use :func:`semver.VersionInfo.parse` instead.
+    .. versionadded:: 2.7.2
+       Added :func:`semver.parse_version_info`
+    :param version: version string
+    :return: a :class:`VersionInfo` instance
+    >>> version_info = semver.VersionInfo.parse("3.4.5-pre.2+build.4")
+    >>> version_info.major
+    3
+    >>> version_info.minor
+    4
+    >>> version_info.patch
+    5
+    >>> version_info.prerelease
+    'pre.2'
+    >>> version_info.build
+    'build.4'
+    """
+    return VersionInfo.parse(version)
+
+
+@deprecated(version="2.10.0")
+def compare(ver1, ver2):
```

Comment:
Due to #258 we should maybe keep this function.

Reply:
Where should we put it? I'm not sure about `__init__.py`

Reply:
I'm going to kick this down the road. I'm not personally convinced that we should undeprecated this yet.